tests/unit_tests/db_engine_specs/test_databend.py

```diff
@@ -0,0 +1,145 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from datetime import datetime
+from typing import Any, Dict, Optional, Type
+from unittest.mock import Mock
+
+import pytest
+from sqlalchemy.types import (
+    Boolean,
+    Date,
+    DateTime,
+    DECIMAL,
+    Float,
+    Integer,
+    String,
+    TypeEngine,
+)
+
+from superset.utils.core import GenericDataType
+from tests.unit_tests.db_engine_specs.utils import (
+    assert_column_spec,
+    assert_convert_dttm,
+)
+from tests.unit_tests.fixtures.common import dttm
+
+
+@pytest.mark.parametrize(
+    "target_type,expected_result",
+    [
+        ("Date", "to_date('2019-01-02')"),
+        ("DateTime", "to_dateTime('2019-01-02 03:04:05')"),
+        ("UnknownType", None),
+    ],
+)
+def test_convert_dttm(
+    target_type: str, expected_result: Optional[str], dttm: datetime
+) -> None:
+    from superset.db_engine_specs.databend import DatabendEngineSpec as spec
+
+    assert_convert_dttm(spec, target_type, expected_result, dttm)
+
+
+def test_execute_connection_error() -> None:
+    from urllib3.exceptions import NewConnectionError
+
+    from superset.db_engine_specs.databend import DatabendEngineSpec
+    from superset.db_engine_specs.exceptions import SupersetDBAPIDatabaseError
+
+    cursor = Mock()
+    cursor.execute.side_effect = NewConnectionError(
+        "Dummypool", "Exception with sensitive data"
+    )
+    with pytest.raises(SupersetDBAPIDatabaseError) as ex:
+        DatabendEngineSpec.execute(cursor, "SELECT col1 from table1")
+
+
+@pytest.mark.parametrize(
+    "target_type,expected_result",
+    [
+        ("Date", "to_date('2019-01-02')"),
+        ("DateTime", "to_dateTime('2019-01-02 03:04:05')"),
+        ("UnknownType", None),
+    ],
+)
+def test_connect_convert_dttm(
+    target_type: str, expected_result: Optional[str], dttm: datetime
+) -> None:
+    from superset.db_engine_specs.databend import DatabendEngineSpec as spec
+
+    assert_convert_dttm(spec, target_type, expected_result, dttm)
```

Comment:
This looks like a duplicate of `test_convert_dttm` on line 50?

Reply:
Aha, that's it. Thanks!